src/components/Notifications/EventLog/usePrimaryToolbarFilterConfigWrapper.tsx

```diff
@@ -0,0 +1,234 @@
+            );
+
+            return {
+                bundleId: bundleName,
+                category: bundleDisplayName || `${bundleName} Loading...`,
+                chips: applicationChips
+            };
+        };
+
+        const applicationsByBundle: { [key: string]: string[] } = {};
+        (filters.application as string[])?.forEach(queryParam => {
+            const nameSplit = queryParam.split('.');
+            const bundleName = nameSplit[0];
+            const application = nameSplit[1];
+
+            if (!applicationsByBundle[bundleName]) {
+                applicationsByBundle[bundleName] = [ application ];
+            } else {
+                applicationsByBundle[bundleName].push(application);
+            }
+        });
+
+        const partialBundleFilters = Object.keys(applicationsByBundle).map(bundleName => {
+            const bundle = bundles?.find(bundle => bundle.name === bundleName);
+            const applications = bundle?.children?.filter(application => applicationsByBundle[bundleName].includes(application.name));
+            return createCustomFilterObj(bundleName, bundle, applications);
+        });
+
+        const completeBundleFilters = (filters.bundle as string[])?.map(queryParam => {
+            const bundle = bundles?.find(bundle => bundle.name === queryParam);
+            return createCustomFilterObj(queryParam, bundle, bundle?.children);
+        }) || [];
+
+        return completeBundleFilters.concat(partialBundleFilters) as EventLogCustomFilter[];
+    }, []);
+
+    // Run ONLY twice
+    //     1st - setup loading placeholders for filters
+    //     2nd - render actual filter bundles/applications
+    React.useEffect(() => {
+        setCustomFilters(mapToEventLogCustomFilter(filters, bundles.length !== 0 ? bundles : undefined));
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [ bundles, mapToEventLogCustomFilter ]);
+
+    const applicationFilter = React.useMemo(() => {
+        return {
+            label: 'Application',
+            type: 'custom',
+            filterValues: {
+                children: <EventLogTreeFilter
+                    groups={ bundles }
+                    placeholder={ 'Filter by Application' }
+                    filters={ customFilters }
+                    updateFilters={ setCustomFilters }
+                />
+            }
+        } as any;
+    }, [ bundles, customFilters ]);
+
+    const activeFiltersConfig = React.useMemo(() => {
+        const activeFilters = toolbarConfig.activeFiltersConfig.filters as EventLogCustomFilter[];
+        const nonCustomFilters = activeFilters.filter(activeFilter => activeFilter && !activeFilter.bundleId);
+        return nonCustomFilters.concat(customFilters);
+    }, [ customFilters, toolbarConfig.activeFiltersConfig.filters ]);
+
+    // Update URL Query Params for Bundles
+    setFilters.bundle(produce(prev => {
+        if (bundles.length === 0) {
+            return filters.bundle;
```

Comment:
Is this meant to be a no-op?  You can return `undefined` to mark "nothing happened"

Reply:
You are correct that is supposed to be a case where nothing changes... will change to `return` to better symbolize that, even though I think technically as it is now, it does the same thing functionally.